ports/rp2/CMakeLists.txt

```diff
@@ -159,6 +159,27 @@ set(PICO_SDK_COMPONENTS
     tinyusb_device
 )
 
+if (MICROPY_PY_NETWORK)
```

Comment:
Doing it like this requires `MICROPY_PY_NETWORK` to be enabled in the mpconfigboard.cmake file for a board.

Since `modusocket.c` and `modnetwork.c` already guard themselves with the relevant `#if`'s, it should work the same to just unconditionally include all these source files in the build, and rely on settings in `mpconfigboard.h` to enable networking.

I think it's simpler to enable things in .h rather than .cmake (or .mk) files.  Would you agree?

Reply:
Yes I agree, but in that case they should be added to extmod/extmod.cmake.

Reply:
Yes, sounds good.